File: P6-NascarAngularPage/src/app/componentes/pilotos/LecturaPilotoDTO.ts
import { LecturaMarcaNombreDTO } from "../LecturaMarcaDTO";

export interface LecturaPilotoDTO {
    id: number;
    nombre: string;
    numero: string | null;
    fotoPiloto: string;
    carrerasGanadas: number;
    poles: number;
    top5s: number;
    top10s: number;
    edad: number;
    campeonatos: number;
    enActivo: boolean;
    auto: LecturaAutoDTO;
    nacionalidad: LecturaNacionalidadDTO;
}

export interface PatchPilotoDTO {
    nombre: string;
    numero: string | null;
    carrerasGanadas: number;
    poles: number;
    top5s: number;
    top10s: number;
    campeonatos: number;
    enActivo: boolean;
    nacionalidadId: number;
    edad: number;
}

export interface CrearPilotoDTO extends PatchPilotoDTO {
    fotoPiloto: File | null | undefined;
}

export interface LecturaAutoDTO {
    id: number;
    piloto: LecturaNombrePilotoDTO;
    foto: string;
    marca: LecturaMarcaNombreDTO;
}

export interface LecturaNacionalidadDTO {
    id: number;
    nombre: string;
    bandera: string;
}

export interface LecturaNombrePilotoDTO {
    id: number;
    nombre: string;
    numero: string | null;
}

File: P5-NascarSeasonWebAPI/NascarPage/NascarPage/DTOs/LecturaPilotoDTO.cs
using NascarPage.Entitys;

namespace NascarPage.DTOs
{
    public class LecturaPilotoDTO
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;
        public string? Numero { get; set; }
        public string FotoPiloto { get; set; } = null!;
        public int CarrerasGanadas { get; set; }
        public int Poles { get; set; }
        public int Top5s { get; set; }
        public int Top10s { get; set; }
        public int Campeonatos { get; set; }
        public bool enActivo { get; set; }
        public int Edad { get; set; }
        public LecturaAutoDTO Auto { get; set; } = null!;
        public LecturaNacionalidadDTO Nacionalidad { get; set; } = null!;
    }

    public class LecturaNombrePilotoDTO
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;
        public string? Numero { get; set; }
    }
}
